tests/test_scalers.py

```diff
@@ -24,3 +25,25 @@ def test_exponential(value, base):
         assert result == 0
     else:
         assert math.copysign(1, result) == math.copysign(1, value)
+
+
+real_halves = floats(allow_nan=False, allow_infinity=False, width=16)
+
+
+@given(
+    value=real_halves,
+    input_range=tuples(real_halves, real_halves).filter(lambda x: x[0] != x[1]),
+    output_range=tuples(real_halves, real_halves)
+    .map(sorted)
+    .filter(lambda x: x[0] < x[1]),
```

Comment:
Should the function check that these conditions are actually met?

Reply:
We could definitely do with documenting the function's preconditions in some way. Whether these are the function's intended preconditions is another question, but they probably are.